src/components/ClusterManager.js

```diff
@@ -0,0 +1,520 @@
+  const { cpu: currentCpu, memory: currentMemory } = machineTypesByName[machineType]
+  return div([
+    div({ style: styles.row }, [
+      div({ style: { ...styles.col1, ...styles.label } }, 'CPUs'),
+      div({ style: styles.col2 }, [
+        readOnly ?
+          currentCpu :
+          h(Select, {
+            wrapperStyle: styles.smallSelect,
+            searchable: false,
+            clearable: false,
+            value: currentCpu,
+            onChange: ({ value }) => onChangeMachineType(_.find({ cpu: value }, machineTypes).name),
+            options: _.map(cpu => ({ label: cpu, value: cpu }), _.uniq(_.map('cpu', machineTypes)))
+          })
+      ]),
+      div({ style: { ...styles.col3, ...styles.label } }, 'Disk size'),
+      div([
+        readOnly ?
+          diskSize :
+          h(IntegerInput, {
+            style: styles.smallInput,
+            min: 100,
+            max: 64000,
+            value: diskSize,
+            onChange: onChangeDiskSize
+          }),
+        ' GB'
+      ])
+    ]),
+    div({ style: styles.row }, [
+      div({ style: { ...styles.col1, ...styles.label } }, 'Memory'),
+      div({ style: styles.col2 }, [
+        readOnly ?
+          currentMemory :
+          h(Select, {
+            wrapperStyle: styles.smallSelect,
+            searchable: false,
+            clearable: false,
+            value: currentMemory,
+            onChange: ({ value }) =>
+              onChangeMachineType(_.find({ cpu: currentCpu, memory: value }, machineTypes).name),
+            options: _.map(
+              ({ memory }) => ({ label: memory, value: memory }),
+              _.sortBy('memory', _.filter({ cpu: currentCpu }, machineTypes))
+            )
+          }),
+        ' GB'
+      ])
+    ])
+  ])
+}
+
+const ClusterIcon = ({ shape, onClick, disabled }) => {
+  return h(Interactive, {
+    style: { color: onClick && !disabled && Style.colors.secondary },
+    as: 'div',
+    onClick: disabled ? undefined : onClick
+  }, [icon(shape, { size: 20, class: 'is-solid' })])
+}
+
+export default class ClusterManager extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      open: false,
+      busy: false,
+      clusters: undefined,
+      profile: 'moderate',
+      masterMachineType: 'n1-standard-4',
+      masterDiskSize: 500,
+      jupyterUserScriptUri: '',
+      numberOfWorkers: 0,
+      numberOfPreemptibleWorkers: 0,
+      workerDiskSize: 500,
+      workerMachineType: 'n1-standard-4'
+    }
+  }
+
+  refreshClusters = async () => {
+    const { namespace } = this.props
+    const allClusters = await Leo.clustersList()
+    const clusters = _.filter({
+      googleProject: namespace,
+      creator: getBasicProfile().getEmail()
+    }, allClusters)
+    this.setState({ clusters })
+  }
+
+  componentDidMount() {
+    this.refreshClusters()
+    this.interval = setInterval(this.refreshClusters, 3000)
```

Comment:
Should this interval be running in the cases when no change is expected (only one cluster is "active")?

Reply:
It would certainly be possible to limit the updates, but this way it will even catch unexpected changes, e.g. on other open browser windows, and do the right thing. Ultimately, this would be a really nice place to have some sort of push notification capability.

Reply:
Could this be more like every 30-60 seconds when the main cluster wasn't last seen in a transition state?